src/vrapp/VRVolumeApp.cpp

```diff
@@ -546,89 +541,94 @@ void VRVolumeApp::render(const MinVR::VRGraphicsState &render_state)
   // overwrite MV for 2D viewing
   if (m_is2d)
-    m_model_view = m_trackball.get_view_matrix();
+    m_view_matrix = m_trackball.get_view_matrix();
 
   glMatrixMode(GL_PROJECTION);
   glLoadMatrixf(glm::value_ptr(m_projection_mtrx));
 
+  glm::mat4 general_model_view = glm::scale(glm::mat4(1.0), glm::vec3(m_ui_view->get_scale(), m_ui_view->get_scale(), m_ui_view->get_scale()));
+
   // setup Modelview for volumes
   for (int i = 0; i < m_volumes.size(); i++)
   {
     for (int j = 0; j < m_volumes[i].size(); j++)
     {
-      glm::mat4 tmp = m_model_view;
-      tmp = tmp * m_object_pose;
-      tmp = glm::scale(tmp, glm::vec3(m_ui_view->get_scale(), m_ui_view->get_scale(), m_ui_view->get_scale() * m_ui_view->get_z_scale()));
+      glm::mat4 volume_matrix = m_view_matrix * m_object_pose * glm::scale(general_model_view, glm::vec3(1.0, 1.0, m_ui_view->get_z_scale()));
       if (!m_animated)
-        tmp = glm::translate(tmp, glm::vec3(m_volumes[i][j]->get_volume_position().x, m_volumes[i][j]->get_volume_position().y, m_volumes[i][j]->get_volume_position().z));
-      m_volumes[i][j]->set_volume_mv(tmp);
+      {
+        volume_matrix = glm::translate(volume_matrix, glm::vec3(m_volumes[i][j]->get_volume_position().x, m_volumes[i][j]->get_volume_position().y, m_volumes[i][j]->get_volume_position().z));
+      }
+      m_volumes[i][j]->set_volume_mv(volume_matrix);
     }
   }
 
   // setup Modelview for meshes
-  glm::mat4 volume_mv;
-  if (m_mesh_model)
+  for (Mesh *mesh : m_mesh_models)
   {
-    int i = 0;
-    if (m_volumes.size() > m_models_volumeID[i])
-    {
-      volume_mv = m_volumes[0][m_models_volumeID[i]]->get_volume_mv();
-      volume_mv = glm::translate(volume_mv, glm::vec3(-0.5f, -0.5f, -0.5f * m_volumes[0][m_models_volumeID[i]]->get_volume_scale().x / (m_volumes[0][m_models_volumeID[i]]->get_volume_scale().z)));
-      volume_mv = glm::scale(volume_mv, glm::vec3(m_volumes[0][m_models_volumeID[i]]->get_volume_scale().x, m_volumes[0][m_models_volumeID[i]]->get_volume_scale().y, m_volumes[0][m_models_volumeID[i]]->get_volume_scale().x));
 
-      m_mesh_model->setMVMatrix(volume_mv);
-      m_models_MV[i] = volume_mv;
-    }
+    Volume *volume = m_volumes[mesh->get_volume_id()][0];
+    glm::mat4 volume_mv = volume->get_volume_mv();
+    float px = volume->get_volume_scale().x;
+    float py = volume->get_volume_scale().y;
+    float pz = volume->get_volume_scale().z;
+
+    /*
+    * TO DO: Fix parent child relationship to not affect the mesh with z-scale
+    // glm::vec3 volume_position = volume_mv[3];
+    // glm::mat4 mesh_model_matrix =  glm::translate(general_model_view, volume_position);
+     //volume_mv = glm::translate(general_model_view, volume_position);
+     //glm::mat4 mesh_model_matrix = glm::translate(volume_mv, glm::vec3(-0.5f, -0.5f, -0.5f * px / pz));
+     ////mesh_model_matrix = glm::scale(mesh_model_matrix, glm::vec3(px, py, px));
+    **/
```

Comment:
Add GitHub issue to the TODO and the `//`s can be deleted since they're inside the multiline comment already

Reply:
👍